llvm/lib/Transforms/IPO/RepoPruning.cpp

```diff
@@ -192,6 +239,20 @@ static void deleteGlobalObject(GlobalObject *GO) {
   }
 }
 
+static void removeOrPrune(GlobalObject &GO) {
```

Comment:
Added, thanks.

Reply:
I should probably have been more clear but by “documentation” I meant Doxygen-style documentation as per the other two new functions here (hasExistingGOInRepository() and hasExistingGOInModule()).

When you say ”prune or remove” what determines which of the two happens? (Answer in the Doxygen comments rather than in this conversation!)

Reply:
Thanks Paul. 
Doxygen-style documentation is added.